### PR TITLE
feat(create-discord-bot): bun/deno templates

### PR DESCRIPTION
**Pull request by iCrawl**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**

### CODE REVIEW

File: packages/create-discord-bot/src/helpers/packageManager.ts [hunk 1]
```diff
@@ -47,19 +47,52 @@ export function resolvePackageManager(): PackageManager {
...
-			installCommand = `${packageManager} install`;
+			console.log();
 			break;
-		case 'yarn':
-			installCommand = packageManager;
+		case 'bun':
+			console.log();
+			break;
+		case 'npm':
 			break;
```
Comment (jaw0r3k) [inline packages/create-discord-bot/src/helpers/packageManager.ts:76]:
```suggestion
```

its definitely not necessary

File: packages/create-discord-bot/src/helpers/packageManager.ts [hunk 2]
```diff
@@ -47,19 +47,52 @@ export function resolvePackageManager(): PackageManager {
...
+			installCommand = `${packageManager} cache --reload src/index.ts`;
+			break;
 		case 'pnpm':
-			installCommand = `${packageManager} install`;
+			console.log();
 			break;
-		case 'yarn':
-			installCommand = packageManager;
+		case 'bun':
+			console.log();
```
Comment (jaw0r3k) [inline packages/create-discord-bot/src/helpers/packageManager.ts:73]: Do we really need console.log in every case?

File: packages/create-discord-bot/template/Deno/src/events/ready.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Events } from 'npm:discord.js@^14.13.0';
+import type { Event } from './index.ts';
+
+export default {
+	name: Events.ClientReady,
+	once: true,
+	async execute(client) {
+		console.log(`Ready! Logged in as ${client.user.tag}`);
+	},
+} satisfies Event<'ready'>;
```
Comment (jaw0r3k) [inline packages/create-discord-bot/template/Deno/src/events/ready.ts:10]:
```suggestion
} satisfies Event<Events.ClientReady>;
```

File: packages/create-discord-bot/template/Bun/src/events/ready.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Events } from 'discord.js';
+import type { Event } from './index.ts';
+
+export default {
+	name: Events.ClientReady,
+	once: true,
+	async execute(client) {
+		console.log(`Ready! Logged in as ${client.user.tag}`);
+	},
+} satisfies Event<'ready'>;
```
Comment (jaw0r3k) [inline packages/create-discord-bot/template/Bun/src/events/ready.ts:10]:
```suggestion
} satisfies Event<Events.ClientReady>;
```

File: packages/create-discord-bot/src/helpers/packageManager.ts [hunk 5]
```diff
@@ -47,19 +47,52 @@ export function resolvePackageManager(): PackageManager {
...
 	switch (packageManager) {
-		case 'npm':
+		case 'yarn':
+			console.log();
+			installCommand = [
+				`${packageManager} set version stable`,
+				`${packageManager} config set nodeLinker node-modules`,
+				`${packageManager} config set logFilters --json '[{ "code": "YN0002", "level": "discard" }, { "code": "YN0013", "level": "discard" }, { "code": "YN0032", "level": "discard" }, { "code": "YN0060", "level": "discard" }]'`,
+				`${packageManager} plugin import interactive-tools`,
+				`${packageManager} plugin import workspace-tools`,
```
Comment (jaw0r3k) [inline packages/create-discord-bot/src/helpers/packageManager.ts:62]:
```suggestion
				`yarn set version stable`,
				`yarn config set nodeLinker node-modules`,
				`yarn config set logFilters --json '[{ "code": "YN0002", "level": "discard" }, { "code": "YN0013", "level": "discard" }, { "code": "YN0032", "level": "discard" }, { "code": "YN0060", "level": "discard" }]'`,
				`yarn plugin import interactive-tools`,
				`yarn plugin import workspace-tools`,
```

just cleaner code

File: packages/create-discord-bot/template/Deno/.vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	"editor.defaultFormatter": "denoland.vscode-deno",
+	"editor.formatOnSave": true,
+	"editor.codeActionsOnSave": {
+		"source.fixAll": true,
+		"source.organizeImports": false
+	},
+	"editor.trimAutoWhitespace": false,
+	"files.insertFinalNewline": true,
+	"files.eol": "\n",
+	"npm.packageManager": "[REPLACE_ME]",
```
Comment (jaw0r3k) [inline packages/create-discord-bot/template/Deno/.vscode/settings.json:11]: image

Yeah...

File: packages/create-discord-bot/src/helpers/packageManager.ts (hunk 2 above)
Comment (iCrawl) [reply packages/create-discord-bot/src/helpers/packageManager.ts:73]: It's only for some to create proper spacing in the terminal output.

Comment (jaw0r3k) [reply packages/create-discord-bot/src/helpers/packageManager.ts:73]: One console.log before switch statement isnt enough? Without duplicating it in every case

File: packages/create-discord-bot/src/helpers/packageManager.ts (hunk 5 above)
Comment (iCrawl) [reply packages/create-discord-bot/src/helpers/packageManager.ts:62]: I don't think it matters too much does it, the packageManager here cannot be anything different anyway

File: packages/create-discord-bot/template/Deno/.vscode/settings.json (hunk 6 above)
Comment (iCrawl) [reply packages/create-discord-bot/template/Deno/.vscode/settings.json:11]: 😠

File: packages/create-discord-bot/src/helpers/packageManager.ts [hunk 7]
```diff
@@ -47,19 +47,52 @@ export function resolvePackageManager(): PackageManager {
  * @param packageManager - The package manager to use
  */
 export function install(packageManager: PackageManager) {
-	let installCommand;
+	let installCommand: string[] | string = `${packageManager} install`;
+
+	console.log(`Installing dependencies with ${packageManager}...`);
 
 	switch (packageManager) {
```
Comment (jaw0r3k) [inline packages/create-discord-bot/src/helpers/packageManager.ts:54]: just change it to 
if (packageManager === 'yarn')

and use one console.log

Comment (iCrawl) [reply packages/create-discord-bot/src/helpers/packageManager.ts:54]: I guess thats true.

Comment (iCrawl) [reply packages/create-discord-bot/src/helpers/packageManager.ts:54]: But I also want to use the console log only on yarn, pnpm, and deno. I can make a default case and get rid of one statement and make a default instead.

**Pull request merged**